spacy/lang/ru/lemmatizer.py

```diff
@@ -28,43 +28,50 @@ def __init__(
                 from pymorphy2 import MorphAnalyzer
             except ImportError:
                 raise ImportError(
-                    "The Russian lemmatizer mode 'pymorphy2' requires the "
-                    "pymorphy2 library. Install it with: pip install pymorphy2"
+                    "The lemmatizer mode 'pymorphy2' requires the "
+                    "pymorphy2 library and dictionaries. Install them with: "
+                    "pip install pymorphy2"
+                    "# for Ukrainian dictionaries:"
+                    "pip install pymorphy2-dicts-uk"
                 ) from None
             if getattr(self, "_morph", None) is None:
-                self._morph = MorphAnalyzer()
-        elif mode == "pymorphy3":
+                self._morph = MorphAnalyzer(lang="ru")
+        elif mode in {"pymorphy3", "pymorphy3_lookup"}:
             try:
                 from pymorphy3 import MorphAnalyzer
             except ImportError:
                 raise ImportError(
-                    "The Russian lemmatizer mode 'pymorphy3' requires the "
-                    "pymorphy3 library. Install it with: pip install pymorphy3"
+                    "The lemmatizer mode 'pymorphy3' requires the "
+                    "pymorphy3 library and dictionaries. Install them with: "
+                    "pip install pymorphy3"
+                    "# for Ukrainian dictionaries:"
+                    "pip install pymorphy3-dicts-uk"
                 ) from None
             if getattr(self, "_morph", None) is None:
-                self._morph = MorphAnalyzer()
+                self._morph = MorphAnalyzer(lang="ru")
         super().__init__(
             vocab, model, name, mode=mode, overwrite=overwrite, scorer=scorer
         )
 
-    def pymorphy2_lemmatize(self, token: Token) -> List[str]:
+    def _pymorphy_lemmatize(self, token: Token) -> List[str]:
         string = token.text
         univ_pos = token.pos_
         morphology = token.morph.to_dict()
         if univ_pos == "PUNCT":
             return [PUNCT_RULES.get(string, string)]
         if univ_pos not in ("ADJ", "DET", "NOUN", "NUM", "PRON", "PROPN", "VERB"):
-            # Skip unchangeable pos
-            return [string.lower()]
+            return self.pymorphy2_lookup_lemmatize(token)
```

Comment:
```suggestion
            return self._pymorphy_lookup_lemmatize(token)
```

Reply:
really disheartening to see you throwing away this refactor

Reply:
It's not meant to be disheartening. It is really just a practical/technical decision so that we can review the changes to the lemmatizer algorithm properly as it shows up here: https://github.com/explosion/spaCy/pull/11811/files. If people look at the git history in the future, they can also see directly how the algorithm was modified.

As I said above, we're not opposed to the additional refactoring (and now I think I'd put it in `spacy.lang.xx` for v3 or `spacy.lang.mul` for v4), it just needs to be in a separate, future PR.